relays/ethereum-client/src/client.rs

```diff
@@ -53,6 +56,23 @@ impl Client {
 	pub fn reconnect(&mut self) {
 		self.client = Self::build_client(&self.params);
 	}
+}
+
+impl Client {
+	/// Returns true if client is connected to at least one peer and is in synced state.
+	pub async fn ensure_synced(&self) -> Result<()> {
+		match Ethereum::syncing(&self.client).await? {
+			SyncState::NotSyncing => Ok(()),
+			SyncState::Syncing(syncing) => {
+				let missing_headers = syncing.highest_block.saturating_sub(syncing.current_block);
+				if missing_headers > MAJOR_SYNC_BLOCKS.into() {
```

Comment:
There is probably a status call for ETH node as well, but it's not really worth implementing given we are not actively working on the PoA bridge.